pkg/component/worker/config/config.go

```diff
@@ -0,0 +1,114 @@
+/*
+Copyright 2022 k0s authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
+	"encoding/json"
+	"fmt"
+	"reflect"
+
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	kubeletv1beta1 "k8s.io/kubelet/config/v1beta1"
+
+	"go.uber.org/multierr"
+	"sigs.k8s.io/yaml"
+)
+
+type Profile struct {
+	KubeletConfiguration kubeletv1beta1.KubeletConfiguration
+}
+
+func (p *Profile) DeepCopy() *Profile {
+	if p == nil {
+		return nil
+	}
+	out := new(Profile)
+	p.DeepCopyInto(out)
+	return out
+}
+
+func (p *Profile) DeepCopyInto(out *Profile) {
+	*out = *p
+	p.KubeletConfiguration.DeepCopyInto(&out.KubeletConfiguration)
+}
+
+func (p *Profile) Validate(path *field.Path) (errs field.ErrorList) {
```

Comment:
Looks like this function does nothing. If it's needed to implement an interface, it could be just `return nil`

Reply:
Removed.